ert/data/record/_transformation.py

```diff
@@ -227,3 +229,74 @@ async def _save_record_to_file(record: Record, location: Path, mime: str) -> Non
     else:
         async with aiofiles.open(str(location), mode="wb") as fb:
             await fb.write(record.data)  # type: ignore
+
+
+class RecordTreeTransformation(RecordTransformation):
+    """
+    Write all leaf records of a RecordTree to individual files.
+    """
+
+    def __init__(self, sub_path: Optional[str] = None):
+        if sub_path is not None:
+            raise NotImplementedError
+
+    async def transform_input(
+        self,
+        record: Record,
+        mime: str,
+        runpath: Path,
+        location: Path,
+    ) -> None:
+        if isinstance(record, NumericalRecordTree):
```

Comment:
Early raise?

Reply:
Changed to `if not isinstance ... raise TypeError` (if that was what you meant?)